src/app/modules/shared_models/link_preview_item.nim

```diff
@@ -4,6 +4,7 @@ import ../../../app_service/service/message/dto/link_preview
 type
   Item* = ref object
     unfurled*: bool
+    immutable*: bool
```

Comment:
`immutable` describes the model item behaviour. `removed` describes the row state and the behaviour needs to be deduced by reading the implementation. Eg. It's not clear for a `removed` row if it can be added back, why it's still in the model or if it can be updated after being removed.

For `immutable` I find it clear by reading the public API that it's a row that can't be updated.